storage/testonly/admin_storage_tester.go

```diff
@@ -126,25 +127,199 @@ func (tester *AdminStorageTester) TestCreateTree(t *testing.T) {
+			updateFunc: validLogFunc,
+			wantTree:   &validLog,
+		},
+		{
+			desc:       "invalidLog",
+			createTree: &referenceLog,
+			updateFunc: invalidLogFunc,
+			wantErr:    true,
+		},
+		{
+			desc:       "readonlyChanged",
+			createTree: &referenceLog,
+			updateFunc: readonlyChangedFunc,
+			wantErr:    true,
+		},
+		{
+			desc:       "validMap",
+			createTree: &referenceMap,
+			updateFunc: validMapFunc,
+			wantTree:   &validMap,
+		},
+	}
+	for _, test := range tests {
+		createdTree, err := createTree(s, test.createTree)
+		if err != nil {
+			t.Errorf("createTree() = (_, %v), want = (_, nil)", err)
+			continue
+		}
+
+		updatedTree, errOnUpdate, err := updateTree(s, createdTree.TreeId, test.updateFunc)
+		if err != nil && !errOnUpdate {
+			t.Errorf("%v: updateTree() failed with non-Update error: %v", test.desc, err)
+			continue
+		}
+
+		if hasErr := err != nil; hasErr != test.wantErr {
+			t.Errorf("%v: updateTree() = (_, %v), wantErr = %v", test.desc, err, test.wantErr)
+			continue
+		} else if hasErr {
+			continue
+		}
+
+		if createdTree.TreeId != updatedTree.TreeId {
+			t.Errorf("%v: TreeId = %v, want = %v", test.desc, updatedTree.TreeId, createdTree.TreeId)
+		}
+		if createdTree.CreateTimeMillisSinceEpoch != updatedTree.CreateTimeMillisSinceEpoch {
+			t.Errorf("%v: CreateTime = %v, want = %v", test.desc, updatedTree.CreateTimeMillisSinceEpoch, createdTree.CreateTimeMillisSinceEpoch)
+		}
+		if createdTree.UpdateTimeMillisSinceEpoch > updatedTree.UpdateTimeMillisSinceEpoch {
+			t.Errorf("%v: UpdateTime = %v, want >= %v", test.desc, updatedTree.UpdateTimeMillisSinceEpoch, createdTree.UpdateTimeMillisSinceEpoch)
+		}
+		// Copy storage-generated values to wantTree before comparing
+		wantTree := *test.wantTree
+		wantTree.TreeId = updatedTree.TreeId
+		wantTree.CreateTimeMillisSinceEpoch = updatedTree.CreateTimeMillisSinceEpoch
+		wantTree.UpdateTimeMillisSinceEpoch = updatedTree.UpdateTimeMillisSinceEpoch
+		if !reflect.DeepEqual(updatedTree, &wantTree) {
+			t.Errorf("%v: updatedTree doesn't match wantTree:\n"+
+				"got =  %v,\n"+
+				"want = %v", test.desc, updatedTree, &wantTree)
 		}
+
+		if storedTree, err := getTree(s, updatedTree.TreeId); err != nil {
+			t.Errorf(":%v: getTree() = (_, %v), want = (_, nil)", test.desc, err)
+		} else if !reflect.DeepEqual(storedTree, updatedTree) {
+			t.Errorf("%v: storedTree doesn't match updatedTree:\n"+
+				"got =  %v,\n"+
+				"want = %v", test.desc, storedTree, updatedTree)
+		}
+
+		if unrelatedAfterTest, err := getTree(s, unrelatedTree.TreeId); err != nil {
+			t.Errorf("%v: getTree() = (_, %v), want = (_, nil)", test.desc, err)
+		} else if !reflect.DeepEqual(unrelatedAfterTest, unrelatedTree) {
+			t.Errorf("%v: unrelatedTree changed:\n"+
+				"got  = %v,\n"+
+				"want = %v", test.desc, unrelatedAfterTest, unrelatedTree)
+		}
+	}
+}
+
+func createTree(s storage.AdminStorage, tree *trillian.Tree) (*trillian.Tree, error) {
+	ctx := context.Background()
```

Comment:
(nit, as it's a test; x3) Be nice to stick the recommended habit of passing `ctx` as first parameter, so that `content.Background()` is only ever called in `main` and `TestThing` top-levels.